pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -209,7 +209,7 @@ public synchronized void channelInactive(ChannelHandlerContext ctx) throws Excep
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
         super.exceptionCaught(ctx, cause);
-        LOG.warn("[{}] Got exception {} : Message: {} State: {}", remoteAddress, cause.getClass().getSimpleName(),
+        LOG.warn("[{}] Got exception {} : Message: {} State: {} Cause: {}", remoteAddress, cause.getClass().getSimpleName(),
                 cause.getMessage(), state,
                 ClientCnx.isKnownException(cause) ? null : cause);
```

Comment:
```suggestion
```suggestion
        if (ClientCnx.isKnownException(cause)) {
            LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                    remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state);
        } else {
            LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                    remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state, cause);
        }
```

To stop the IDEA complaining, you can make this change.
```

Reply:
Or:

```java
        LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state,
                ClientCnx.filterKnownException(cause));

// in ClientCnx.java

    public static Throwable filterKnownException(Throwable t) {
        return isKnownException(t) ? null : t;
    }
```

It's somehow IDEA shortcoming.

Reply:
> Or:
> 
> ```java
>         LOG.warn("[{}] Got exception {} : Message: {} State: {}",
>                 remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state,
>                 ClientCnx.filterKnownException(cause));
> 
> // in ClientCnx.java
> 
>     public static Throwable filterKnownException(Throwable t) {
>         return isKnownException(t) ? null : t;
>     }
> ```
> 
> It's somehow IDEA shortcoming.

@tisonkun You are right, I am misled by idea complain.